parmed/openmm/parameters.py

```diff
@@ -178,10 +179,16 @@ def _write_omm_atom_types(self, dest):
         dest.write(' </AtomTypes>\n')
 
     def _write_omm_residues(self, dest):
+        if not self.residues: return
         dest.write(' <Residues>\n')
         for name, residue in iteritems(self.residues):
             if not isinstance(residue, ResidueTemplate):
                 continue
+            # if AtomTypes are available, this will throw an exception if the
+            # residue is claiming a Type that doesn't exist
+            if self.atom_types:
```

Comment:
Ah, I see from the commit message.  Maybe we can make `typeify_templates` a little smarter about only "typeifying" what it can and not complaining when some types don't exist?

Also, maybe we should get rid of `ResidueTemplateContainer` instances (basically the solvent boxes) when calling `OpenMMParameterSet.from_parameterset` so that we only *keep* residues that we explicitly plan on writing to XML files.  What do you think about that?